src/test/unit/io/deserializer_stdvector_test.cpp

```diff
@@ -175,6 +175,50 @@ TEST(deserializer_array, sum_to_zero) {
   }
 }
 
+TEST(deserializer_array, sum_to_zero_matrix) {
```

Comment:
I would like to see a boundary test for a 1 x 1 matrix, which has zero underlying unconstrained parameters and always just contains 0.